libs/block_generator/include/json_validator.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef ZKEMV_FRAMEWORK_LIBS_BLOCK_GENERATOR_JSON_VALIDATOR_HPP_
+#define ZKEMV_FRAMEWORK_LIBS_BLOCK_GENERATOR_JSON_VALIDATOR_HPP_
+
+#include <optional>
+#include <sstream>
+#include <string>
+#include <valijson/adapters/boost_json_adapter.hpp>
+#include <valijson/schema.hpp>
+#include <valijson/schema_parser.hpp>
+#include <valijson/validator.hpp>
+
+/// @brief Inline json validator
+/// @returns the first found error if exists
+inline std::optional<std::string> validate_json(const boost::json::value &schema_json,
+                                                const boost::json::value &target_json) {
+    valijson::Schema schema;
+    valijson::SchemaParser schemaParser;
+
+    valijson::adapters::BoostJsonAdapter schemaAdapter(schema_json);
+
+#if VALIJSON_USE_EXCEPTIONS
+    try {
+#endif
+        schemaParser.populateSchema(schemaAdapter, schema);
+#if VALIJSON_USE_EXCEPTIONS
+    } catch (...) {
```

Comment:
Can we catch an explicit exception here and get a meaningful message from it? Is `valijson` cool enough to tell you what is wrong and where?

Reply:
`valijson` is definitely not flexible in terms of error handling, you can probably see it in error messages in tests. But it is the only validator I found that provide interface for `boost::json`. 
In this case, I just haven't found any documentation about possible exceptions here. But I have examined the implementation now and it look like that `std::exception` message covers all the cases. So added it.